### ISSUE TITLE
ServiceFabric :  A cluster id has not been configured, BUG

### ISSUE THREAD

**Issue by hanxinimm** [dotnet/orleans#4226]
what's error with me?
how to do it?

```
var listener = OrleansServiceListener.CreateStateless(
                (serviceContext, builder) =>
                {
                    // Optional: use Service Fabric for cluster membership.
                     builder.UseServiceFabricClustering(serviceContext);

                    var config = new ClusterConfiguration();

                    config.Globals.ReminderServiceType = ReminderServiceProviderType.Disabled;

                    // Service Fabric manages port allocations, so update the configuration using those ports.
                    builder.Configure<EndpointOptions>(options => {
                        options.AdvertisedIPAddress = IPAddress.Loopback;
                        options.ConfigureFromServiceContext(serviceContext);
                    });

                    builder.AddIncomingGrainCallFilter<ExceptionConversionFilter>();

                    // Tell Orleans to use this configuration.
                    builder.UseConfiguration(config);

                    builder.ConfigureServices(ConfigureDelegate);

                    // Add your application assemblies.
                    builder.ConfigureApplicationParts(parts =>
                    {
                        // Alternative: add all loadable assemblies in the current base path (see AppDomain.BaseDirectory).
                        parts.AddFromApplicationBaseDirectory().WithCodeGeneration();
                    });
                });
```

**Comment by ReubenBond** [comment]
You need to call `builder.Configure<ClusterOptions>(options => options.ClusterId = "your id");`

**Comment by hanxinimm** [comment]
@ReubenBond  

it remove  this code 

`builder.UseConfiguration(config);`

it work good

ServiceFabric don't need old method  UseConfiguration？

```
 var listener = OrleansServiceListener.CreateStateless(
                (serviceContext, builder) =>
                {

                    // Optional: configure logging.
                    builder.ConfigureLogging(logging => logging.AddDebug());


                    builder.AddMemoryGrainStorage("MemoryStore");

                    builder.Configure<EndpointOptions>(options =>
                    {
                        options.AdvertisedIPAddress = System.Net.IPAddress.Loopback;
                        options.ConfigureFromServiceContext(serviceContext);
                    });



                    builder.UseServiceFabricClustering(serviceContext);

                    builder.ConfigureServices(ConfigureDelegate);

                    // Add your application assemblies.
                    builder.ConfigureApplicationParts(parts =>
                    {
                        // Alternative: add all loadable assemblies in the current base path (see AppDomain.BaseDirectory).
                        parts.AddFromApplicationBaseDirectory().WithCodeGeneration();
                    });
                });
```

**Comment by ReubenBond** [comment]
That's correct, Service Fabric does not require the old configuration method anymore

**Comment by hanxinimm** [comment]
@ReubenBond 

it not work 

`builder.Configure<ClusterOptions>(options => options.ClusterId = "SeekerApp-StatelessCommandService");`

![image](https://user-images.githubusercontent.com/13499426/37461800-a16b507e-288a-11e8-82cb-866a47626aeb.png)

![image](https://user-images.githubusercontent.com/13499426/37461820-b290025a-288a-11e8-9cb7-315362bebec3.png)

**Comment by ReubenBond** [comment]
This will be fixed in the next release, it was fixed in #4211.
As a work-around, you can configure the endpoints more explicitly, like this:
```C#
builder.Configure<EndpointOptions>(options =>
{
    options.AdvertisedIPAddress = IPAddress.Loopback;
    options.ConfigureFromServiceContext(serviceContext);
    options.SiloListeningEndpoint = new IPEndPoint(IPAddress.Any, options.SiloPort);
    options.GatewayListeningEndpoint = new IPEndPoint(IPAddress.Any, options.GatewayPort);
});
```

**Comment by hanxinimm** [comment]
@ReubenBond 

Both the server and the client need to add this code

server :
```
builder.Configure<ClusterOptions>(options =>
                    {
                        options.ClusterId = serviceContext.ServiceName.ToString();
                    });

```
client:

```
     return new ClientBuilder()
                .UseServiceFabricClustering(new Uri(serviceName))
                .Configure<ClusterOptions>(options => options.ClusterId = serviceName)
                .ConfigureApplicationParts(parts => parts.AddFromApplicationBaseDirectory().WithCodeGeneration())
                .Build();
        
```

it work good
